drivers/sensor/scd30/scd30.c

```diff
@@ -0,0 +1,241 @@
+		return -EIO;
+	}
+
+	/* C02 data */
+	rc = scd30_fill_data_buf(rx_buf, data_buf);
+	if (rc != 0) {
+		return -EIO;
+	}
+	rc = scd30_fill_data_buf(rx_buf + 3, data_buf + 2);
+	if (rc != 0) {
+		return -EIO;
+	}
+
+	/* temperature data */
+	rc = scd30_fill_data_buf(rx_buf + 6, data_buf + 4);
+	if (rc != 0) {
+		return -EIO;
+	}
+	rc = scd30_fill_data_buf(rx_buf + 9, data_buf + 6);
+	if (rc != 0) {
+		return -EIO;
+	}
+
+	/* relative humidity */
+	rc = scd30_fill_data_buf(rx_buf + 12, data_buf + 8);
+	if (rc != 0) {
+		return -EIO;
+	}
+	rc = scd30_fill_data_buf(rx_buf + 15, data_buf + 10);
+	if (rc != 0) {
+		return -EIO;
+	}
+
+	float co2_ppm, temp, rel_hum;
+
+	co2_ppm = scd30_bytes_to_float(data_buf);
+	temp = scd30_bytes_to_float(data_buf + 4);
+	rel_hum = scd30_bytes_to_float(data_buf + 8);
+
+	sensor_value_from_double(&data->co2_ppm, co2_ppm);
+	sensor_value_from_double(&data->temp, temp);
+	sensor_value_from_double(&data->rel_hum, rel_hum);
+
+	return 0;
+}
+
+static int scd30_channel_get(const struct device *dev, enum sensor_channel chan,
+			     struct sensor_value *val)
+{
+	struct scd30_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_CO2:
+		memcpy(val, &data->co2_ppm, sizeof(struct sensor_value));
+		break;
+
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		memcpy(val, &data->temp, sizeof(struct sensor_value));
+		break;
+
+	case SENSOR_CHAN_HUMIDITY:
+		memcpy(val, &data->rel_hum, sizeof(struct sensor_value));
+		break;
+
+	default:
+		return -1;
+	}
+	return 0;
+}
+
+static const struct sensor_driver_api scd30_driver_api = {
+	.sample_fetch = scd30_sample_fetch,
+	.channel_get = scd30_channel_get,
+};
+
+int scd30_write_command(const struct device *dev, uint16_t cmd)
+{
+	const struct scd30_config *cfg = dev->config;
+	uint8_t tx_buf[2];
+
+	sys_put_be16(cmd, tx_buf);
+	return i2c_write_dt(&cfg->bus, tx_buf, sizeof(tx_buf));
+}
+
+static uint8_t scd30_compute_crc(uint8_t *data, uint8_t data_len)
+{
+	return crc8(data, data_len, SCD30_CRC8_POLYNOMIAL, SCD30_CRC8_INIT, false);
+}
+
+int scd30_check_crc(uint8_t *data, uint8_t data_len, uint8_t checksum)
+{
+	uint8_t actual_crc = scd30_compute_crc(data, data_len);
+
+	if (checksum != actual_crc) {
+		LOG_DBG("CRC check failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int scd30_write_register(const struct device *dev, uint16_t cmd, uint16_t val)
+{
+	const struct scd30_config *cfg = dev->config;
+	uint8_t tx_buf[SCD30_CMD_SINGLE_WORD_BUF_LEN];
+
+	sys_put_be16(cmd, &tx_buf[0]);
+	sys_put_be16(val, &tx_buf[2]);
+	tx_buf[4] = scd30_compute_crc(tx_buf, sizeof(tx_buf));
+
+	return i2c_write_dt(&cfg->bus, tx_buf, sizeof(tx_buf));
+}
+
+static int scd30_init(const struct device *dev)
+{
+	LOG_DBG("Initializing SCD30");
+	const struct scd30_config *cfg = dev->config;
+
+	if (!device_is_ready(cfg->bus.bus)) {
+		LOG_DBG("Failed to get pointer to %s device!", log_strdup(cfg->bus.bus->name));
+		return -ENODEV;
+	}
+
+	if (!cfg->bus.addr) {
+		LOG_DBG("No I2C address");
+		return -EINVAL;
+	}
```

Comment:
when does this condition occur?

Reply:
You are right. When an address is not specified, a compiler error is thrown. So this is redundant